books/RayTracingInOneWeekend.html

```diff
@@ -1590,15 +1595,16 @@
 
         std::cout << "P3\n" << image_width << " " << image_height << "\n255\n";
 
-        for (int j = image_height-1; j >= 0; --j) {
-            std::clog << "\rScanlines remaining: " << j << ' ' << std::flush;
+        for (int j = 0; j < image_height; ++j) {
+            std::clog << "\rScanlines remaining: " << (image_height - j) << ' ' << std::flush;
+            auto t = (j + random_double()) / (image_height-1);
+
             for (int i = 0; i < image_width; ++i) {
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++ highlight
                 color pixel_color(0, 0, 0);
                 for (int sample = 0; sample < samples_per_pixel; ++sample) {
-                    auto u = (i + random_double()) / (image_width-1);
-                    auto v = (j + random_double()) / (image_height-1);
-                    ray r = cam.get_ray(u, v);
+                    auto s = (i + random_double()) / (image_width-1);
+                    ray r = cam.get_ray(s, t);
```

Comment:
Did you drop the definition of t?

Reply:
Oh, I see it's up above.
Hmm.

It's more clear when s and t are defined together. But it's a waste to reassign j every iteration of i.

I have a mild preference for how it was previously for clarity.

Reply:
I waffled a bit on this one as well. One point in favor of the constant hoisting is that it's a situation that comes up often in raytracing, where you have scanline loops and pixel loops.

Sort of an uncommented teaching moment, but I also see the advantage of keeping things simpler but a bit slower.